ReactNativeTraining/src/app/screens/Home/screens/Chats/index.js

```diff
@@ -1,10 +1,15 @@
-import React from 'react';
+import { connect } from 'react-redux';
 
-import data from '../../../../../utils/constants';
-import ContactList from '../../../components/ContactList/index';
+import Chats from './layout';
 
-function Chats() {
-  return <ContactList data={data.contacts} />;
-}
+const mapStateToProps = state => ({
+  messages: state.messages.filter(message => message.receiverId !== undefined),
+  contacts: state.contacts.filter(
+    contact =>
+      state.messages.some(message => message.senderId === contact.id || message.receiverId === contact.id) &&
+      contact.id !== 1
+  ),
```

Comment:
This is fine, works fine, but presents performance issues. We need to use here `Reselect` library, as i told u. Every time that any change is realized in state, for example when receive messages, all this filters would be run, and this cause that props change and therefore cause unwanted re-renders. Your contacts doesn't change, and u dont want re-run this filters, but runs every time that state change.

Check this. https://github.com/reactjs/reselect